### DISCUSSION TITLE
System-wide configuration path and filename?

### DISCUSSION THREAD

**Discussion by dirteat** [fvwmorg/fvwm3#1022] [Q&A]
Hi there,
I am getting a bit lost with the new system-wide config file. How it is supposed to work with 1.1.0?

It seems that passing --sysconfdir at build time has no effect for version 1.1.0, system.fvwm3rc seems to be ignored as well (that used to work). Only a file named "config" within --datadir seems to still work.

Are these issues, features? I'd be happy to have any standard, but something not changing too much within releases, that creates quite some issues for distributing system-wide config while packaging fvwm3 on distros.

Thanks,
Chris.

**Comment by somiaj** [comment]
Are you just guessing the name of the system wide configuration has changed? This hasn't been messed with for some time, and should be the same as fvwm2. I would start with the manual page (look for the `INITIALIZATION` section) in which it states that it first runs `Read config`, which basically looks in `$FVMW_USERDIR` and then `$FVWM_DATADIR` for the file `config`, if that file is not found, it looks in some other locations (Note I'm unsure if `/usr/local` here is fixed or just means the install `--prefix`):

```
$HOME/.fvwm/config
/usr/local/share/fvwm/config

$HOME/.fvwm/.fvwm2rc
$HOME/.fvwm2rc
/usr/local/share/fvwm/.fvwm2rc
/usr/local/share/fvwm/system.fvwm2rc
/etc/system.fvwm2rc

Please note, the last 5 locations are not guaranteed to be supported in the future
```

If none of these files are found, then the `default-config` is loaded. This is also well documented on the Wiki:

https://www.fvwm.org/Wiki/Config/Fvwm2rc/

**Comment by dirteat** [comment]
Well, thank you...
If I am asking it is because this does not work anymore, at least for me, and I thought it was an undocumented change, especially after reading the warning in the man. Searching system.fvwm3rc within --sysconfdir was also working, so things have been modified. But if you say nothing has been messed in the sector,  I'll chase the bug and open a bug report.

Thank you.
Chris.

**Comment by somiaj** [comment]
Read a little closer, no where in anything I said said use `/etc/system.fvwm3rc`, use `/etc/system.fvwm2rc`.

**Comment by somiaj** [comment]
I just tested, `/etc/system.fvwm2rc` is correctly used as the configuration file (if none of the other configuration files in the manual page I quoted above are found). Again you are making assumptions, instead of using exactly the file that was documented.

**Comment by dirteat** [comment]
If this some standard of your to be aggressive? Are you reading my thread? Do you know what --sysconfdir means?
Thank you for your pointless answer. If I have an issue I'll open a bug report and send you a patch.

**Comment by somiaj** [comment]
There have been no changes that I know of to how `--sysconfdir` works, so this won't be new with fvwm3. But as for the name of the file you keep stating, there is no `system.fvwm3rc`, and fvwm3 is not looking for any configuration file with `fvwm3rc` in it. The old name which is still supported for backwards compatibility is `system.fvwm2rc`. I only tested that this worked in `/etc`, I haven't tested that you can move this location with `--sysconfdir`.

Looking at the source, it appears this should still be supported,

```
FVWM_CONFDIR='${sysconfdir}'dnl used _only_ to search for system.fvwm2rc
```

Further if you look in `fvwm/fvwm3.c` you can see that it is still supported using:

```
xasprintf(&cfg_loc[++nl], "%s/system%s", FVWM_CONFDIR, FVWM2RC);
```

In short, the very last place that is looked (before the default-config is then used) is `FVWM_CONFDIR/system.fvwm2rc`, which defaults to `/etc/system.fvwm2rc`. Note if any other configuration file is found before this one, then that one will be used instead.

From everything I see and tested, it appears the issue is you have made an incorrect assumption that the file name should now be `system.fvwm3rc` and not `system.fvwm2rc`. I don't think it was pointless to point out this issue in the exact name you were using that I saw.

Also, don't try to infer meaning from text, it is hard to get right. I'm not being aggressive, just trying to point out that you have made a small incorrect assumption, and if you fix that everything I see states it should work (as this has not been changed for a long time and is the same as fvwm2).

**Comment by somiaj** [comment]
I guess it could be possible you are using a modified source that has changed the hard coded definition, `AC_DEFINE(FVWM2RC, ".fvwm2rc")`, but I'm not seeing this modification in the fvwm3 source.
